txn_test.go

```diff
@@ -378,3 +367,128 @@ func (s *TransactionTestSuit) TestExceedMaxRows(c *C) {
 	err := tx.Commit()
 	c.Assert(err, NotNil)
 }
+
+func (s *TransactionTestSuit) TestCheckCommitStatus(c *C) {
+	conf := defaultTxnConf
+	hook := newHook()
+	hook.beforeCommitSecondary = func(txn *themisTxn, ctx interface{}) (bool, interface{}, error) {
+		// add before commit secondary hook, just return, do not commit
+		// secondaries
+		log.Info("before commit secondary")
+		return false, nil, nil
+	}
+	tx := newTxn(s.cli, conf)
+	tx.(*themisTxn).setHook(hook)
+	for i := 0; i < 10; i++ {
+		tx.Put(themisTestTableName, hbase.NewPut([]byte(fmt.Sprintf("%s_%d", getTestRowKey(c), i))).AddValue(cf, q, []byte("test")))
+	}
+	tx.Commit()
+	commitTs := tx.(*themisTxn).commitTs
+
+	pRow := append([]byte(nil), tx.(*themisTxn).primaryRow.row...)
+	sRow := append([]byte(nil), tx.(*themisTxn).secondaryRows[0].row...)
+
+	for i := 0; i < 10; i++ {
+		conf = defaultTxnConf
+		tx = newTxn(s.cli, conf)
+		tx.Put(themisTestTableName, hbase.NewPut([]byte(pRow)).AddValue(cf, q, []byte("newVal")))
+		tx.Commit()
+	}
+
+	hook = newHook()
+	hook.beforePrewriteLockClean = func(txn *themisTxn, ctx interface{}) (bool, interface{}, error) {
+		lock := ctx.(Lock)
+		cc := lock.Primary().Coordinate()
+		exists, err := txn.lockCleaner.IsLockExists(lock.Coordinate(), 0, lock.Timestamp())
+		c.Assert(err, IsNil)
+		c.Assert(exists, Equals, true)
+		ts, err := txn.lockCleaner.GetCommitTimestamp(cc, lock.Timestamp())
+		c.Assert(err, IsNil)
+		c.Assert(ts, Equals, commitTs)
+		return true, nil, nil
+	}
+	tx = newTxn(s.cli, conf)
+	tx.(*themisTxn).setHook(hook)
+	tx.Put(themisTestTableName, hbase.NewPut([]byte(sRow)).AddValue(cf, q, []byte("newVal")))
+	tx.Commit()
+}
+
+func createChoosePrimaryRowHook(target string) *txnHook {
+	hook := newHook()
+	hook.afterChoosePrimaryAndSecondary = func(txn *themisTxn, ctx interface{}) (bool, interface{}, error) {
+		txn.secondary = nil
+		txn.secondaryRows = nil
+		for tblName, rowMutations := range txn.mutationCache.mutations {
+			for _, rowMutation := range rowMutations {
+				row := rowMutation.row
+				for i, mutation := range rowMutation.mutationList(true) {
+					colcord := hbase.NewColumnCoordinate([]byte(tblName), row, mutation.Family, mutation.Qual)
+					// set the first column as primary if primary is not set by user
+					if string(row) == target {
+						txn.primary = colcord
+						txn.primaryRowOffset = i
+						txn.primaryRow = rowMutation
+					} else {
+						txn.secondary = append(txn.secondary, colcord)
+					}
+				}
+				if string(row) != target {
+					txn.secondaryRows = append(txn.secondaryRows, rowMutation)
+				}
+			}
+		}
+		return true, nil, nil
+	}
+	return hook
+}
+
+func (s *TransactionTestSuit) TestPrewriteSecondaryMissingRows(c *C) {
```

Comment:
Please add a few description about the bug.
It is not easy for others to figure out the point.